### PR TITLE
feat(locator.withText): allow narrowing locators to those with text

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: tests/page/locator-query.spec.ts [hunk 1]
```diff
@@ -59,3 +59,23 @@ it('should throw on due to strictness 2', async ({ page }) => {
   const e = await page.locator('option').evaluate(e => {}).catch(e => e);
   expect(e.message).toContain(`strict mode violation`);
 });
+
+it('should filter by text', async ({ page }) => {
```
Comment (dgozman) [inline tests/page/locator-query.spec.ts:63]: Let's add a test with `<div>foo<span>hello world</span>bar</div>` and `locator('div').withText('hello world')`?

File: docs/src/api/class-locator.md [hunk 2]
```diff
@@ -1059,3 +1059,15 @@ orderSent.WaitForAsync();
 
 ### option: Locator.waitFor.state = %%-wait-for-selector-state-%%
 ### option: Locator.waitFor.timeout = %%-input-timeout-%%
+
+## method: Locator.withText
```
Comment (dgozman) [inline docs/src/api/class-locator.md:1063]: Maybe hasText to match css?

File: packages/playwright-core/src/server/supplements/recorder/utils.ts [hunk 3]
```diff
@@ -58,15 +58,3 @@ export function describeFrame(frame: Frame): { frameName?: string, frameUrl: str
     return { isMainFrame: false, frameUrl: frame.url(), frameName: frame.name() };
   return { isMainFrame: false, frameUrl: frame.url() };
 }
-
-export function escapeWithQuotes(text: string, char: string = '\'') {
```
Comment (dgozman) [inline packages/playwright-core/src/server/supplements/recorder/utils.ts:62]: Did you forget to git add a new file?

**Review by dgozman — APPROVED** [review]

File: docs/src/api/class-locator.md (hunk 2 above)
Comment (pavelfeldman) [reply docs/src/api/class-locator.md:1063]: Not sure!

File: packages/playwright-core/src/server/supplements/recorder/utils.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/supplements/recorder/utils.ts:62]: yeah

File: tests/page/locator-query.spec.ts (hunk 1 above)
Comment (pavelfeldman) [reply tests/page/locator-query.spec.ts:63]: I hope this test would fail? We are interested in the outermost node matching the text here.

**Pull request merged**